src/View.php

```diff
@@ -17,22 +21,36 @@
 final class View extends BaseView
 {
     /**
-     * Marks the beginning of a page.
+     * Marks the beginning of a view.
      */
-    public function beginPage(): void
+    public function beginPage(array $parameters = []): void
     {
         ob_start();
         PHP_VERSION_ID >= 80000 ? ob_implicit_flush(false) : ob_implicit_flush(0);
 
-        $this->eventDispatcher->dispatch(new PageBegin($this->getViewFile()));
+        $this->eventDispatcher->dispatch(new PageBegin($this, $parameters));
     }
 
     /**
-     * Marks the ending of a page.
+     * Marks the ending of a view.
      */
-    public function endPage(): void
+    public function endPage(array $parameters = []): void
```

Comment:
I don't think parameters are useful here and in alike methods.

Reply:
Done